### ISSUE TITLE
"Load configuration" does not default to current day's output directory

### ISSUE THREAD

**Issue by philipstarkey** [labscript-suite-oldfinal1/runmanager#1]
**Original report ([archived issue](https://labscript-suite.github.io/labscript-suite-bitbucket-archive/#!/labscript_suite/runmanager/issues/1)) by Shaun Johnstone (Bitbucket: [shjohnst](https://bitbucket.org/%7B652df738-313d-472c-b7b5-a3f8f98a3322%7D/), GitHub: [shjohnst](https://github.com/shjohnst)).**

----------------------------------------
When you load a configuration in runmanager, the output directory is set to the one used on the day the configuration was saved. If runmanager was using the default directory (not manually changed) when it was saved, then it should load the appropriate default directory for the current day.

**Comment by philipstarkey** [comment]
**Original comment by Philip Starkey (Bitbucket: [pstarkey](https://bitbucket.org/%7B48af65db-e5fc-459c-a7eb-52eb1f9a5690%7D/), GitHub: [philipstarkey](https://github.com/philipstarkey)).**

----------------------------------------
I've tried to fix this, but it is extremely difficult. This is mainly because the GTK file chooser behaves very strangely. If you set the path on a file chooser programatically, and then immediately read out the path again, it gives you the previous path, not the one you just set. The path is not updated internally until control is returned to the GTK event loop.

Long story short, this isn't going to be fixed until runmanager is ported to PyQt.

**Comment by philipstarkey** [comment]
**Original comment by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------
Qt is so much nicer than GTK. Although it doesn't have a FileChooser widget, it was easy to make something that looks like one, and setting the text is immediate, like almost everything else.

In fact, I only realised recently that so long as you're in the main thread, slots are called immediately when signals are emitted, rather than being queued up and executing when execution gets back to the event loop. It's fantastic, it means everything is exactly in the state you expect it to be, and you almost never have to queue things up for later once the changes you've made take effect - because they've already taken effect.

Anyway this seems trivial, I'll do it when I'm implementing the save and load functions in the Qt port sometime in the next week.

**Comment by philipstarkey** [comment]
**Original comment by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------

* changed assignee_account_id from "557058:2f99420c-1dbd-4837-952c-82c421b8fbdd" to "557058:cbf1bc43-1dc2-477b-9e25-1a8f40fd7ee3"
* changed assignee from "philipstarkey (Bitbucket: [philipstarkey](https://bitbucket.org/philipstarkey), GitHub: [philipstarkey](https://github.com/philipstarkey))" to "cbillington (Bitbucket: [cbillington](https://bitbucket.org/cbillington), GitHub: [chrisjbillington](https://github.com/chrisjbillington))"

**Comment by philipstarkey** [comment]
**Original comment by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------

* changed state from "new" to "resolved"

Resolved in the Qt port (runmanager 2.0)


The save data now stores a parameter called is_using_default_shot_output_folder.
The loader checks it and if True, sets the current default folder at load time.
